src/calculateRentalCost.js

```diff
@@ -4,7 +4,26 @@
  * @return {number}
  */
 function calculateRentalCost(days) {
-  // write code here
+  const pricePerDay = 40;
+  const periodDiscounts = { 3: 20, 7: 50 };
+
+  const discount = getDiscount(days, periodDiscounts);
+
+  return days * pricePerDay - discount;
+}
+
+/**
+ * @param {number} days
+ * @param {Object} periodDiscounts
+ *
+ * @return {number}
+ */
+function getDiscount(days, periodDiscounts) {
+  const allPeriods = Object.getOwnPropertyNames(periodDiscounts);
+  const viablePeriods = allPeriods.filter((value) => value <= days);
```

Comment:
The variable `viablePeriods` is not necessary. You could directly calculate the discount in this function without filtering the periods.